Fix Javadoc URLs for Java 1.8

sanjayl opened the pull request:
Before submitting a PR make sure the following things have been done:

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [x] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [ ] You've updated the readme (if adding/changing middleware)

Thanks!

An issue where the URLs for Javadoc in Java 1.8 has a different format
than prior Java version, namely, parenthesis surrounding argument lists
have been changed to dashes. Added tests as well.

bbatsov commented on test/clj/cider/nrepl/middleware/info_test.clj:
```diff
@@ -16,6 +16,28 @@
   (:resource (info/file-info x)))
 
+(deftest test-javadoc-url
+  (testing "java 1.6"
```
We don't support Java 6 anymore (although Clojure itself does), so this test is not necessary. Plus, it's the same behaviour as in 7.

Malabarba commented on src/cider/nrepl/middleware/util/java.clj:
```diff
@@ -73,9 +73,14 @@
-   (str (javadoc-url class) "#" member
-        (when argtypes
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (read-string util/java-api-version)
```
I'd rather use `parse-int` or `Integer.`.

sanjayl commented:
Made the changes rec'ed above. Added a few more changes to get as close as possible to the new URL spec (although from what I've surmised, we'll never be 100% since the URLs are created via parsing the source, not via reflection). Changed the unit tests a bit.

Apologizing in advance at the ugliness of this code, I wasn't sure how to go about fixing it since the necessary changes were scattered across the code base.

bbatsov commented on src/cider/nrepl/middleware/util/java.clj:
```diff
@@ -73,9 +73,19 @@
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (Integer/parseInt util/java-api-version)
+         old-format   (< java-version 8)
+         open-paren   (if old-format "(" "-")
```
Let's just have two string templates - one for old and one for the new format and insert into them whatever's needed. Or just build the new old or new string without actually introducing those locals. They serve no purpose and their names are confusing for the new format (e.g. close-paren is a "-"). I'd rather have two functions that build the old/new format or something like this.

Malabarba replied:
Yes, I think it's better to just have one `if` where each branch is the entire logic for that version.

bbatsov replied:
Or one `if` branching out to two helper functions. :-) Anyways, we're on the same page here.

sanjayl commented:
Yeah, that was pretty dumb of me. Here's the redo

bbatsov commented:
Perfect! :+1:
